### ISSUE TITLE
Rule Change: `ignoreClassWithStaticInitBlock` as default behaviour in `no-unused-vars`

### ISSUE THREAD

**Issue by erosman** [eslint/eslint#18980]
### What rule do you want to change?

no-unused-vars

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code
```js
/* eslint no-unused-vars: ["error", { "ignoreClassWithStaticInitBlock": true }] */

class ClassWithSIB {
  static {
    // …
  }
}
```
### What does the rule currently do for this code?

Currently, the option "ignoreClassWithStaticInitBlock" is required to suppress the error.

### What will the rule do after it's changed?

"ignoreClassWithStaticInitBlock" would no longer be needed.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

previously: #17772

[no-unused-vars](https://eslint.org/docs/latest/rules/no-unused-vars) states:

> Variables that are declared and not used anywhere in the code are most likely an error due to incomplete refactoring. **Such variables take up space in the code and can lead to confusion by readers**.
> 
> This rule is aimed at **eliminating unused variables, functions, and function parameters**.

In other words, unused variables (and functions, parameters) are the variables that their removals have no consequence in the execution of the code.

A `class` with a "static initialization block"  does not match the above category since its removal will have an impact in the execution of the code.

Therefore, I believe ignoring the  "static initialization block" in `no-unused-vars` should be the default behaviour of the rule.

**Comment by mdjermanovic** [comment]
I think the current default behavior is fine, as this is not a common pattern in JS, so it's more likely that some code is missing or the class can be removed.

> In other words, unused variables (and functions, parameters) are the variables that their removals have no consequence in the execution of the code.
> 
> A `class` with a "static initialization block" does not match the above category since its removal will have an impact in the execution of the code.

Code from the static block can be extracted into a block statement (`{ ... }`) and the functionality will remain the same. Just the class variable would be removed.

@eslint/eslint-team thoughts about this?

**Comment by erosman** [comment]
> Code from the static block can be extracted into a block statement ({ ... }) and the functionality will remain the same. Just the class variable would be removed.

ECMAScript 2022 (ES13) developed "static initialization block" with the explicit intent for it to be used, which nullifies any argument about not using it in favour of an alternative method.

Furthermore, valid code should not have to be altered to match linter's view. ;)

**Comment by nzakas** [comment]
I agree with @mdjermanovic -- I think the rule's current defaults are correct. As there is already an option that support what you'd like to do, I don't think there's any change to be made here.
